index_batcher.go

```diff
@@ -29,15 +51,10 @@ type IndexBatcher struct {
 	// and adding operations to the batch.
 	lock   sync.Mutex
 	batch  *Batch
-	result *rslt
+	result *error
```

Comment:
You are correct that error is an interface and that pointers are very uncommon; however, in this circumstance, the pointer to error is intentional.  Otherwise there is a race between the scheduler releasing previous calls to Index(), Delete() etc. picking up the result of the previous batch and the batchLoop() firing again and overwriting the result that those operations were supposed to have received.  Basically what's happening is described in this PDF (unless I'm missing something): 
[IndexBatcher.pdf](https://github.com/blevesearch/bleve/files/566710/IndexBatcher.pdf)